R/DMMParam.R

```diff
@@ -0,0 +1,199 @@
+#' DMM clustering
+#' 
+#' Run the \code{DMM} function on a distance matrix within \code{\link{clusterRows}}.
+#' 
+#' @param k An integer indicating the number of clusters, or a list of integers defining the possible number of clusters to choose from.
+#' @param type A string specifying the fit function to use to find the optimal number of clusters. Use in combination with k being a list of integers
+#' @param type A boolean specifying 
+#' @param seed An integer specifying the seed to use when doing the DMM algorithm. Will be random if not set.
+#' 
+#' @inheritParams clusterRows
+#' @param BLUSPARAM A \linkS4class{KmeansParam} object.
+#' @param full Logical scalar indicating whether the clustering statistics from both steps should be returned.
+#' 
+#' @author Basil Courbayre
+#'
+#' @references
+#' Dirichlet multinomial mixtures: generative models for microbial metagenomics.
+#' Ian Holmes, Keith Harris, Christopher Quince
+#' Department of Bioengineering, University of California, Berkeley, California, United States of America.
+#' \url{https://pubmed.ncbi.nlm.nih.gov/22319561/}
+#' 
+#' @details
+#' The DMM algorithm (see references tag and citation below) is commonly used in microbial ecology along with metagenomic & 16S rRNA count data.
+#' Holmes I, Harris K, Quince C. Dirichlet multinomial mixtures: generative models for microbial metagenomics. PLoS One. 2012;7(2):e30126. doi: 10.1371/journal.pone.0030126. Epub 2012 Feb 3. PMID: 22319561; PMCID: PMC3272020.
+#' 
+#' Because of this specificity, the best data we can use for this algorithm is microbiota read counts (for example the Twins data set in the DMM package).
+#' To modify an existing DMMParam object \code{x},
+#' users can simply call \code{x[[i]]} or \code{x[[i]] <- value} where \code{i} is any argument used in the constructor.
+#'
+#' If \code{k=NULL}, the cluster search range will be 1:3.
+#' The default for the fit function is \code{laplace}.
+#' 
+#' @return 
+#' The \code{DMMParam} constructor will return a \linkS4class{DMMParam} object with the specified parameters.
+#'
+#' The \code{clusterRows} method will return a factor of length equal to \code{nrow(x)} containing the cluster assignments.
+#' If \code{full=TRUE}, a list is returned with \code{clusters} (the factor, as above) and \code{objects} 
+#' (a list containing \code{dmm}, the dmm object; \code{k}, the number of clusters; 
+#' \code{prob}, the array of probabilities; and \code{seed} the seed for the dmm).
+#'
+#' @examples
+#' clusterRows(iris[,1:4], DMMParam())
+#' clusterRows(iris[,1:4], DMMParam(k=2))
+#' \dontrun{
+#' example(fl <- system.file(package="DirichletMultinomial", "extdata", "Twins.csv")
+#'         count <- t(as.matrix(read.csv(fl, row.names=1)))
+#'         clusterRows(counts, DMMParam(k=1:3, type="laplace")))
+#' }
```

Comment:
what this example function does? is it necessary?

in line 47, there is counts, in 46 count

Reply:
It may not be necessary but it's an example Leo gave to get data from DMM doc. I put it so that people can easily get data to try the function with relevant data.
Should I remove it?

Reply:
It's ok, I'm just not sure about this example() --> if other examples in bluster package do not have that, you can remove it and leave the code inside it